exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/easy/EasyGroupScan.java

```diff
@@ -333,21 +338,73 @@ public boolean canPushdownProjects(List<SchemaPath> columns) {
   }
 
   @JsonProperty
+  @JsonIgnore(value = false)
   public TupleMetadata getSchema() {
     return getTableMetadata().getSchema();
   }
 
   @Override
-  @JsonIgnore
-  public TableMetadata getTableMetadata() {
-    if (tableMetadata == null) {
-      tableMetadata = metadataProvider.getTableMetadata();
+  public AnalyzeInfoProvider getAnalyzeInfoProvider() {
+    return new AnalyzeFileInfoProviderImpl(formatPlugin.getName());
+  }
+
+  @Override
+  protected GroupScanWithMetadataFilterer<?> getFilterer() {
+    return new EasyGroupScanFilterer(this);
+  }
+
+  @Override
+  protected FileTableMetadataProviderBuilder<?> tableMetadataProviderBuilder(MetadataProviderManager source) {
+    if (source.usesMetastore()) {
+      return new MetastoreFileTableMetadataProvider.Builder<>((MetastoreMetadataProviderManager) source);
+    } else {
+      return defaultTableMetadataProviderBuilder(source);
     }
-    return tableMetadata;
   }
 
   @Override
-  public TableMetadataProvider getMetadataProvider() {
-    return metadataProvider;
+  protected FileTableMetadataProviderBuilder<?> defaultTableMetadataProviderBuilder(MetadataProviderManager source) {
+    return new SimpleFileTableMetadataProvider.Builder(source);
+  }
+
+  /**
+   * Implementation of GroupScanWithMetadataFilterer which uses {@link EasyGroupScan} as source and
+   * builds {@link EasyGroupScan} instance with filtered metadata.
+   */
+  private static class EasyGroupScanFilterer extends GroupScanWithMetadataFilterer<EasyGroupScanFilterer> {
+
+    EasyGroupScanFilterer(EasyGroupScan source) {
+      super(source);
+    }
+
+    @Override
+    public AbstractGroupScanWithMetadata<?> build() {
+      EasyGroupScan newScan = new EasyGroupScan((EasyGroupScan) source);
+      newScan.tableMetadata = tableMetadata;
+      // updates common row count and nulls counts for every column
+      if (newScan.getTableMetadata() != null && files != null && newScan.getFilesMetadata().size() != files.size()) {
+        newScan.tableMetadata = TableMetadataUtils.updateRowCount(newScan.getTableMetadata(), files.values());
+      }
+      newScan.partitions = partitions;
+      newScan.segments = segments;
+      newScan.files = files;
+      newScan.matchAllMetadata = matchAllMetadata;
+      newScan.nonInterestingColumnsMetadata = nonInterestingColumnsMetadata;
+
+      newScan.fileSet = newScan.getFilesMetadata().keySet();
+      newScan.selection = FileSelection.create(null, new ArrayList<>(newScan.fileSet), newScan.selectionRoot);
+      try {
+        newScan.initFromSelection(newScan.selection, newScan.formatPlugin);
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to initialize scan from the selection.", e);
+      }
+
+      return newScan;
+    }
+
+    @Override
+    protected EasyGroupScanFilterer self() {
+      return this;
+    }
```

Comment:
Seems odd. What does this do?

Reply:
Cool. I could really use this feature. Can you point me to an example of where you've used it?